sympy/utilities/misc.py

```diff
@@ -440,3 +441,24 @@ def ordinal(num):
     else:
         suffix = 'th'
     return str(n) + suffix
+
+
+def unchanged(func, *args):
```

Comment:
I think it would be better to put this in expr.py. The utilities are pretty much pure-Python.

And although I think the change you made is ok, and the one that @oscarbenjamin suggested in the issue is a known one (where the intent is just to see that the expression is unchanged), if you look at the issue I referenced you will see there is a lot of subtlety here and it may be difficult to know if the `_eval_Eq` method was being tested or if a no-change of expression was being tested.

Well, I guess it isn't that hard. You would have to check the issue page #2942, but I think if a do-nothing tests involves an object that does not have an `_eval_Eq` method, then the test truly is a do-nothing and `unchanged` would be a better choice. A search for `def _eval_Eq` shows the following:
```
Search "def _eval_Eq" (10 hits in 6 files)
  C:\Users\leslie\sympy\sympy\core\numbers.py (1 hits)
	Line 3187:     def _eval_Eq(self, other):
  C:\Users\leslie\sympy\sympy\functions\elementary\integers.py (3 hits)
	Line 149:     def _eval_Eq(self, other):
	Line 233:     def _eval_Eq(self, other):
	Line 341:     def _eval_Eq(self, other):
  C:\Users\leslie\sympy\sympy\matrices\expressions\matexpr.py (1 hits)
	Line 552:     def _eval_Eq(self, other):
  C:\Users\leslie\sympy\sympy\matrices\immutable.py (1 hits)
	Line 77:     def _eval_Eq(self, other):
  C:\Users\leslie\sympy\sympy\polys\rootoftools.py (1 hits)
	Line 957:     def _eval_Eq(self, other):
  C:\Users\leslie\sympy\sympy\sets\sets.py (3 hits)
	Line 621:     def _eval_Eq(self, other):
	Line 970:     def _eval_Eq(self, other):
	Line 1534:     def _eval_Eq(self, other):
```
I don't see complexes on there, so `re(x) == re(x)` is not testing a `re._eval_Eq` method.

Reply:
Oh sorry! I didn't see your comment. I will make the changes accordingly

Reply:
@smichr I have made the changes in the `test_complexes.py` file earlier. 
See this https://github.com/sympy/sympy/blob/2fd29ff5dd5283c3756407decb03ece9340a7943/sympy/functions/elementary/tests/test_complexes.py#L36
and 
https://github.com/sympy/sympy/blob/2fd29ff5dd5283c3756407decb03ece9340a7943/sympy/functions/elementary/tests/test_complexes.py#L132

Reply:
Also, is there a way of checking something like https://github.com/sympy/sympy/pull/2942/files#diff-6a9fda5f073235dfa5f7af5c74cdadddL326 using `unchanged`. I don't think that is possible as of now